client.go

```diff
@@ -348,6 +350,11 @@ func NewClient(cfg *ClientConfig) (cl *Client, err error) {
 		},
 	}
 
+	if cfg.LocalServiceDiscovery.Enabled { 
```

Comment:
I haven't checked this elsewhere, but Enabled internal fields are always suspect to me. Can the config not be optional and/or an enabled field be external?